File: src/mathematics/TrailingZeroFactOne.java
package mathematics;

import java.util.Scanner;

//It's Naive solution
//TC- 0(n) & AC-0(1)
public class TrailingZeroFactOne {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int res = trailZeroFact(n);
        System.out.println(res);
    }
    static int trailZeroFact(int n) {
        int count = 0;
        int fact = 1;
        for(int i = 2; i <= n; i++) {
            fact*=i;
        }
        while(fact%10 == 0) {
            count++;
            fact/=10;
        }
        return count;
    }
}

File: Mathematics/TrailingZeroFactOne.cpp
#include<bits/stdc++.h>
using namespace std;

int trailZeroFact(int n) {
    int fact = 1;
    for(int i = 2; i <= n; i++) {
        fact = fact * i;
    }
    int count = 0;
    while(fact%10 == 0) {
        count++;
        fact=fact/10;
    }
    return count;
}

int main() {
    int n; cin>>n;

    int res = trailZeroFact(n);
    cout<<res<<endl;

    return 0;
}